PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Doctor/DoctorService.cs

```diff
@@ -0,0 +1,106 @@
+using FluentValidation;
+using FluentValidation.Results;
+using PrivateHospitals.Application.Dtos.Doctor;
+using PrivateHospitals.Application.Dtos.Patient;
+using PrivateHospitals.Application.Dtos.User;
+using PrivateHospitals.Application.Interfaces.Doctor;
+using PrivateHospitals.Application.Responses;
+using PrivateHospitals.Core.Models;
+using PrivateHospitals.Data.Interfaces.Doctor;
+using PrivateHospitals.Data.Interfaces.Patient;
+using PrivateHospitals.Data.Interfaces.User;
+
+namespace PrivateHospitals.Application.Services.Doctor;
+
+public class DoctorService: IDoctorService
+{
+    private readonly IDoctorRepository _doctorRepository;
+    private readonly IValidator<RegisterDto> _registerDtoValidator;
+    private readonly IValidator<RegisterDoctorDto> _registerDoctorDtoValidator;
+    private readonly IUserRepository _userRepository;
+
+    public DoctorService(IDoctorRepository doctorRepository, IValidator<RegisterDto> registerDtoValidator, IValidator<RegisterDoctorDto> registerDoctorDtoValidator, IUserRepository userRepository)
+    {
+        _doctorRepository = doctorRepository;
+        _registerDtoValidator = registerDtoValidator;
+        _registerDoctorDtoValidator = registerDoctorDtoValidator;
+        _userRepository = userRepository;
+    }
+
+
+    public async Task<ServiceResponse<bool>> RegisterDoctorAsync(RegisterDoctorDto doctorDto)
+    {   
+        //Register to Doctors
+        ValidationResult validationDoctorResult = await _registerDoctorDtoValidator.ValidateAsync(doctorDto);
+        
+        if (!validationDoctorResult.IsValid)
+        {
+            var errors = validationDoctorResult.Errors
+                .Select(error => error.ErrorMessage)
+                .ToList();
+
+            return ServiceResponse<bool>.ErrorResponse(errors);
+        }
```

Comment:
Move validation to mentioned middleware pipeline

Reply:
Fixed